src/__tests__/index.test.js

```diff
@@ -187,6 +187,75 @@ describe('denormalize', () => {
     expect(denormalize('123', article, entities)).toMatchSnapshot();
   });
 
+  it('denormalizes arrays of entities normalized using Union schemas)', () => {
+    const admin = {
+      type: 'admin',
+      id: 1,
+      name: 'Mr. Admin'
+    };
+    const visitor = {
+      type: 'visitor',
+      id: 2,
+      name: 'Ms Visitor'
+    };
+    const adminSchema = new schema.Entity('admin');
+    const visitorSchema = new schema.Entity('visitor');
+    const userSchema = new schema.Union({
+      admin: adminSchema,
+      visitor: visitorSchema
+    }, 'type');
+    const data = [ admin, visitor ];
+    const { result, entities } = normalize(data, new schema.Array(userSchema));
```

Comment:
Instead of using `userSchema = new schema.Union`, you should only be using `schema.Array`:

```
const userListSchema = new schema.Union({
  admin: adminSchema,
  visitor: visitorSchema
}, 'type');

const { result, entities } = normalize(data, userListSchema);

const denormalized = denormalize(result, userListSchema, entities);
```

The reason that `[ unionSchema ]` works is because plain array syntax does not allow polymorphic mappings, while `schema.Array` does. 

Internally, if you use `schema.Array` with a `schema.Union` as its internal schema, you end up nesting polymorphic mappings, so resolving the schema for a single object in the input array (`result`, in the test code) resolves to a mapping of multiple potential types. `schema.Array` has this mapping potential already built-in to itself.